app/models/shipit/stack.rb

```diff
@@ -168,9 +168,13 @@ def schedule_merges
     end
 
     def next_commit_to_deploy
-      commits_to_deploy = commits.order(id: :asc).newer_than(last_deployed_commit).reachable.preload(:statuses)
-      commits_to_deploy = commits_to_deploy.limit(maximum_commits_per_deploy) if maximum_commits_per_deploy
-      commits_to_deploy.to_a.reverse.find(&:deployable?)
+      return @next_commit_to_deploy if defined?(@next_commit_to_deploy)
```

Comment:
A nice ruby way to do this is to do:
```ruby
def next_commit_to_deploy
  @next_commit_to_deploy ||= begin
    ...
  end
end
```

😄

Reply:
I thought about using `||=` however in this case if there is no commit to deploy it will return `nil` and it won't be cached because `nil` is going to be false, that's why I decided to use `defined?`, by using it I can "cache" false values too.

I've noticed some tests failed because of this cache, I could change the signature of this method to `next_commit_to_deploy(cache: true)` but I'm not convinced it is a good approach, do you have a better suggestion?